e2e/Tests/helpers/TestModuleCallbackHandler.cs

```diff
@@ -54,9 +55,11 @@ internal async Task SetModuleReceiveMethodAndRespondAsync<T>(object moduleRespon
                     {
                         request.MethodName.Should().Be(ExpectedDirectMethodRequest.MethodName, "The expected method name should match what was sent from service");
 
-                        var expectedRequestPayload = (T)ExpectedDirectMethodRequest.Payload;
-                        request.TryGetPayload(out T actualRequestPayload).Should().BeTrue();
-                        actualRequestPayload.Should().BeEquivalentTo(expectedRequestPayload, "The expected method data should match what was sent from service");
+                        byte[] expectedRequestPayload = ExpectedDirectMethodRequest.Payload;
+                        string expectedRequestPayloadString = Encoding.UTF8.GetString(expectedRequestPayload);
+                        request.TryGetPayload(out byte[] actualRequestPayload).Should().BeTrue();
+                        string jsonString = Encoding.UTF8.GetString(actualRequestPayload);
+                        jsonString.Should().BeEquivalentTo(expectedRequestPayloadString, "The expected method data should match what was sent from service");
```

Comment:
Similar to https://github.com/Azure/azure-iot-sdk-csharp/pull/3340/files#r1267389575